block/internal/executing/executor.go

```diff
@@ -0,0 +1,644 @@
+package executing
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/evstack/ev-node/block/internal/cache"
+	"github.com/evstack/ev-node/block/internal/common"
+	coreexecutor "github.com/evstack/ev-node/core/execution"
+	coresequencer "github.com/evstack/ev-node/core/sequencer"
+	"github.com/evstack/ev-node/pkg/config"
+	"github.com/evstack/ev-node/pkg/genesis"
+	"github.com/evstack/ev-node/pkg/signer"
+	"github.com/evstack/ev-node/pkg/store"
+	"github.com/evstack/ev-node/types"
+)
+
+// broadcaster interface for P2P broadcasting
+type broadcaster[T any] interface {
+	WriteToStoreAndBroadcast(ctx context.Context, payload T) error
+}
+
+// Executor handles block production, transaction processing, and state management
+type Executor struct {
+	// Core components
+	store     store.Store
+	exec      coreexecutor.Executor
+	sequencer coresequencer.Sequencer
+	signer    signer.Signer
+
+	// Shared components
+	cache   cache.Manager
+	metrics *common.Metrics
+
+	// Broadcasting
+	headerBroadcaster broadcaster[*types.SignedHeader]
+	dataBroadcaster   broadcaster[*types.Data]
+
+	// Configuration
+	config  config.Config
+	genesis genesis.Genesis
+	options common.BlockOptions
+
+	// State management
+	lastState    types.State
+	lastStateMtx *sync.RWMutex
+
+	// Channels for coordination
+	txNotifyCh chan struct{}
+	errorCh    chan<- error // Channel to report critical execution client failures
+
+	// Logging
+	logger zerolog.Logger
+
+	// Lifecycle
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+// NewExecutor creates a new block executor.
+// The executor is responsible for:
+// - Block production from sequencer batches
+// - State transitions and validation
+// - P2P broadcasting of produced blocks
+// - DA submission of headers and data
+func NewExecutor(
+	store store.Store,
+	exec coreexecutor.Executor,
+	sequencer coresequencer.Sequencer,
+	signer signer.Signer,
+	cache cache.Manager,
+	metrics *common.Metrics,
+	config config.Config,
+	genesis genesis.Genesis,
+	headerBroadcaster broadcaster[*types.SignedHeader],
+	dataBroadcaster broadcaster[*types.Data],
+	logger zerolog.Logger,
+	options common.BlockOptions,
+	errorCh chan<- error,
+) (*Executor, error) {
+	if signer == nil {
+		return nil, errors.New("signer cannot be nil")
+	}
+
```

Comment:
In the old code, the there was a check for `signer.GetAddress()` equals `genesis.ProposerAddress` for each block creation. 
I think it would be good to check in the constructor instead so that we fail fast on a missmatch

Reply:
https://github.com/evstack/ev-node/pull/2644/commits/7a5e9246b68a07bf77a26d53820e15e83caa31df